dev/src/main/java/org/eclipse/codewind/intellij/ui/module/CodewindModuleBuilder.java

```diff
@@ -55,55 +70,42 @@ public ModuleWizardStep getCustomOptionsStep(WizardContext context, Disposable p
             if (!InstallUtil.getInstallStatus().isInstalled()) {
                 return new InstallCodewindStep();
             }
+
             if (!InstallUtil.getInstallStatus().isStarted()) {
                 return new StartCodewindStep();
             }
-            return createNewCodewindProjectStep();
+
+            // The custom options step's #onWizardFinished() method doesn't get called, so we can't use
+            // this step to do any validation when the user clicks 'finish'.  Since we only have one real
+            // step to do the validation, that step has to be created in #createWizardSteps.
+            // We still need to provide a custom option step so the user can select the JDK for the project.
+            return new ModuleWizardStep() {
```

Comment:
Are you saying you want to do validation before the user clicks finished? Maybe try overriding ModuleWizardStep's onStepLeaving() method?  This is called before the step is committed and before onWizardFinished.